File: dotnet/Multiplayer/Matchmaking/MatchmakingDirectJoinResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking
{
    /// <summary>
    /// Response returned when a player successfully joins a matchmaking lobby directly.
    /// </summary>
    public class MatchmakingDirectJoinResponse : ApiResponse<EMatchmakingJoinError>
    {
        /// <summary>
        /// Confirmation message for the join operation.
        /// </summary>
        public string Message { get; set; } = string.Empty;

        /// <summary>
        /// The ID of the matchmaking lobby that was joined.
        /// </summary>
        public string Matchmaking_id { get; set; } = string.Empty;
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/MatchmakingDirectJoinResponse.java
package com.michitai.multiplayer.matchmaking;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response returned when a player directly joins a matchmaking lobby.
 */
public class MatchmakingDirectJoinResponse extends ApiResponse {
    @JsonProperty("message")
    private String message;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}
